pkg/controller/federatedtypeconfig/controller.go

```diff
@@ -233,6 +235,16 @@ func (c *Controller) reconcile(qualifiedName util.QualifiedName) util.Reconcilia
 		c.stopController(statusKey, statusStopChan)
 	}
 
+	c.lock.RLock()
+	version, ok := c.targetVersions[typeConfig.Name]
+	c.lock.RUnlock()
+	if ok && typeConfig.Spec.TargetType.Version != version {
```

Comment:
Based on my test, in this `reconcile` function, `ObservedGeneration` and `Generateion` are always equal.

Reply:
Can you confirm that this is still true?  Since the controller is responsible for updating `status.ObservedGeneration`, it logically follows that they 2 fields will not always match.